### PR TITLE
Fix a bug where an element with automatic `height` and specified `max-height` caused subsequent elements to be clamped incorrectly.

### PR DESCRIPTION
**Pull request by foobear**
Whenever an element with undefined/auto `height` would be clamped, it causes subsequent elements to be clamped improperly, resulting in an incorrectly placed ellipsis and overfull contents.

This resolves issue #2 and is an alternative suggestion to PR #5 which introduced extra paint cycles (factor of 2 or 3).

Here is what I suggest: If we encounter a problematic element that does not even meet its maximum dimensions, we explicitly fixate them, since it should never reach them anyway. That way, clamping can happen as before, with the same number of paint cycles (each one is a performance impact).

I am not 100% satisfied with that approach, as it feels hacky. I tried making this work by only improving `getInnerPosition` (which is called on the clamped node) but failed.

Please try this approach in the project you initially encountered the issue and see if it works.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -56,4 +56,4 @@ DEPENDENCIES
   uglifier
 
 BUNDLED WITH
-   1.10.6
+   2.1.4
```
Comment (makmic) [inline Gemfile.lock:56]: Bundler 2 requires Ruby 2.3+. Please update either the `.ruby-version` to >= 2.3 or leave the BUNDLED WITH section untouched.. new devs can't bundle with this combination.

File: test/index.html [hunk 2]
```diff
@@ -14,6 +14,13 @@
       height: 200px;
       max-width: 800px;
       padding: 5px 25px 10px 80px;
+      line-height: 18px;
+    }
+    .clamp-me.-max-height {
+      /*width: 400px;*/
```
Comment (makmic) [inline test/index.html]: Stale comment

**Review by makmic — APPROVED** [review]
Thank you @foobear !

The code code looks fine to me:
* The performance does not seem to degrade compared to `master`
* All edge cases described in the [other PR](https://github.com/makandra/superclamp/pull/5) seem are covered by your code
* I just tested the project where the error occurred originaly, and it works like a charm
* All browsers that I tested are able to clamp the test page as expected (mac/Safari, win7/IE11, Win10/Edge 18|80/FF74/Chrome, ubuntu/Chrome... and some mobile devices)

File: Gemfile.lock (hunk 1 above)
Comment (foobear) [reply Gemfile.lock:56]: Thanks for pointing that out, I had manually switched to 2.3, I believe.
I suggest upgrading to a recent Ruby and Bundler, and upgrade the Gemfile.lock where possible.

**Comment by makmic** [comment]
I think we are ready to merge these changes into master, what do you think? :slightly_smiling_face:

**Comment by foobear** [comment]
Released as 0.2.3
https://www.npmjs.com/package/superclamp
